benchmark/benchmark.py

```
@@ -141,6 +157,100 @@ def _stop_resource_monitor(self, process):
+                    for child in children:
+                        if 'perf' in child.name():
+                            child.send_signal(signal.SIGINT)
+                            child.wait()
+                    # Then kill the flamegraph process (parent)
+                    process.send_signal(signal.SIGINT)
+                    process.wait()
+
+                except psutil.NoSuchProcess:
+                    log.warning(f"Process {process.pid} no longer exists")
+
+                # Generate inverted flamegraph if scripts path is provided and perf.data exists
+                if self.flamegraph_scripts_path and os.path.exists("perf.data"):
+                    self._generate_inverted_flamegraph()
+
+                # Clean up perf.data file if it exists
+                try:
+                    if os.path.exists("perf.data"):
+                        os.remove("perf.data")
+                        log.debug("Cleaned up perf.data file")
+                except Exception:
+                    log.warning("Failed to clean up perf.data file", exc_info=True)
+
+        except Exception:
+            log.error("Error shutting down monitoring:", exc_info=True)
+
+    def _generate_inverted_flamegraph(self):
+        """Generate inverted flamegraph using flamegraph.pl scripts.
+
+        Possible enhancement/TODO: We could optimise runtime here by not using cargo flamegraph to get perf data and compute both graphs from one invocation of stackcollapse, etc.
+        """
+        try:
+            stackcollapse_script = os.path.join(self.flamegraph_scripts_path, "stackcollapse-perf.pl")
+            flamegraph_script = os.path.join(self.flamegraph_scripts_path, "flamegraph.pl")
+
+            # Check if scripts exist
+            if not os.path.exists(stackcollapse_script) or not os.path.exists(flamegraph_script):
+                log.warning(f"Flamegraph scripts not found in {self.flamegraph_scripts_path}")
```

Isn't it better to error out before running benchmarks, than silently ignore it after running them?

---

Sure, lets come back to this as this is only for Mountpoint for now. We need to get this working for other benchmarks.